### DISCUSSION TITLE
Type error with a module called metadata-interface.js with no export of ResolvingViewport

### DISCUSSION THREAD

**Discussion by RestartDK** [vercel/next.js#57606] [Help]
### Summary

I tried deploying my Next 13 application with app router to vercel but I received the following error:

`Type error: Module '"next/dist/lib/metadata/types/metadata-interface.js"' has no exported member 'ResolvingViewport'.`

This seems to happen after the `src/app/page.tsx` is trying to be compiled:

Does anyone know any way to resolve this issue?

**Comment by mass2527** [comment]
Can i see your `src/app/page.tsx`?

**Reply by RestartDK** [reply]
Here is the code for the `page.tsx` file.

```

import Link from "next/link";

import { CreatePost } from "~/app/_components/create-post";
import { api } from "~/trpc/server";

export default function Home() {
  //const recipe = await api.recipe.getAll.query();

  return (
    <main className="flex min-h-screen flex-col items-center justify-center bg-gradient-to-b from-[#2e026d] to-[#15162c] text-white">
      <div className="container flex flex-col items-center justify-center gap-12 px-4 py-16 ">
        <h1 className="text-5xl font-extrabold tracking-tight sm:text-[5rem]">
          <span className="text-green-600">Nourish</span><span className="text-yellow-400">Pal</span>
        </h1>
        <div className="grid grid-cols-1 gap-4 sm:grid-cols-2 md:gap-8">
          <Link
            className="flex max-w-xs flex-col gap-4 rounded-xl bg-white/10 p-4 hover:bg-white/20"
            href="https://create.t3.gg/en/usage/first-steps"
            target="_blank"
          >
            <h3 className="text-2xl font-bold">First Steps →</h3>
            <div className="text-lg">
              Just the basics - Everything you need to know to set up your
              database and authentication.
            </div>
          </Link>
          <Link
            className="flex max-w-xs flex-col gap-4 rounded-xl bg-white/10 p-4 hover:bg-white/20"
            href="https://create.t3.gg/en/introduction"
            target="_blank"
          >
            <h3 className="text-2xl font-bold">Documentation →</h3>
            <div className="text-lg">
              Learn more about Create T3 App, the libraries it uses, and how to
              deploy it.
            </div>
          </Link>
        </div>
        <div className="flex flex-col items-center gap-2">
          <p className="text-2xl text-white">
            yooooo
          </p>
        </div>

        <CrudShowcase />
      </div>
    </main>
  );
}

async function CrudShowcase() {
  const latestPost = await api.recipe.getLatest.query();

  return (
    <div className="w-full max-w-xs">
      {latestPost ? (
        <p className="truncate">Your most recent post: {latestPost.preparation_steps}</p>
      ) : (
        <p>You have no posts yet.</p>
      )}

      <CreatePost />
    </div>
  );
}

```

**Reply by mass2527** [reply]
I believe this may be a Next.js 14-related issue, as [ResolvingViewport was introduced in Next.js 14](https://github.com/vercel/next.js/pull/57302/files#diff-cddbe27fbb7f2869a664909ffb579a3fe45275b84fbd7f5800268dd77ab05fd6R52).

May I ask which version of Next.js you are currently using?

**Reply by RestartDK** [reply]
I am using next 14.0.0.

**Reply by tolgaouz** [reply]
i'm having this issue with next@13.5.1 too. Were you able to find a solution?

**Reply by homemade-ramen** [reply]
Hi

I'm facing a similar problem and was wondering if you could help.

I'm using Next.js version 13.5.6 in my project. When I run the next build in my local environment, it completes successfully. However, when I try the same build process in the GitHub Actions environment, it fails to build.

What's particularly confusing is that the log output shows "Next.js 14.0.1", even though I'm using version 13.5.6:

Here's the log output I get when trying to build in the GitHub Actions environment:
```
$ next build
   ▲ Next.js 14.0.1

   Creating an optimized production build ...
 ✓ Compiled successfully
   Linting and checking validity of types ...
Failed to compile.

src/app/page.tsx
Type error: Module '"next/dist/lib/metadata/types/metadata-interface.js"' has no exported member 'ResolvingViewport'.

error: script "build" exited with code 1 (SIGHUP)
Error: Process completed with exit code 1.
```

**Reply by homemade-ramen** [reply]
The root cause of the problem is that a library called [next-themes](https://github.com/pacocoursey/next-themes) installs Next.js version 14.0.1 by default, and this version of Next.js is used during the build process. This version mismatch is causing the build to fail.

To resolve this, we can use the overrides feature in our package.json file to specify the desired version of Next.js. This will ensure that GitHub Actions uses the correct version during the build process, which will ultimately solve the problem.

As a temporary solution, I recommend implementing this approach. It should help us resolve the Next.js version mismatch issue and ensure successful builds in GitHub Actions.

```json
{
  "dependencies": {
    "next": "13.5.6"
  },
  "overrides": {
    "next": "13.5.6"
  }
}
```
